File: leetcode/find-all-k-distant-indices-in-an-array.cpp
class Solution {
public:
    vector<int> findKDistantIndices(vector<int>& a, int v, int k) {
        vector<int> z;
        for (int i = 0; i < a.size(); i++)
        {
            for (int j = max(i - k, 0); j <= min(i + k, int(a.size()) - 1); j++)
            {
                if (a[j] == v)
                {
                    z.push_back(i);
                    break;
                }
            }
        }
        return z;
    }
};

File: leetcode/find-all-k-distant-indices-in-an-array.py
class Solution:
    def findKDistantIndices(self, a: List[int], key: int, k: int) -> List[int]:
        z = []
        for i in range(len(a)):
            if key in a[max(i-k,0):i+1+k]:
                z.append(i)
        return z